src/scipp/ndimage/__init__.py

```diff
@@ -0,0 +1,247 @@
+    if set(index) != set(x.dims):
+        raise ValueError(f"Data has dims={x.dims} but input argument '{name}' provides "
+                         f"values for {tuple(index)}")
+    return [_delta_to_positional(x, dim, index[dim], dtype=dtype) for dim in x.dims]
+
+
+@_ndfilter
+def gaussian_filter(x: Union[Variable, DataArray],
+                    /,
+                    *,
+                    sigma,
+                    order=0,
+                    **kwargs) -> Union[Variable, DataArray]:
+    """
+    Multidimensional Gaussian filter.
+
+    This is a wrapper around :py:func:`scipy.ndimage.gaussian_filter`. See there for
+    full argument description. There are two key differences:
+
+    - This wrapper uses explicit dimension labels in the ``sigma`` and ``order``
+      arguments. For example, instead of ``sigma=[4, 6]`` use
+      ``sigma={{'time':4, 'space':6}}``
+      (with appropriate dimension labels for the data).
+    - Coordinate values can be used (and should be preferred) for ``sigma``. For
+      example, instead of ``sigma=[4, 6]`` use
+      ``sigma={{'time':sc.scalar(5.0, unit='ms'), 'space':sc.scalar(1.2, unit='mm')}}``.
+      In this case it is required that the corresponding coordinates exist and form a
+      "linspace", i.e., are evenly spaced.
+
+    Warning
+    -------
+    When ``sigma`` is an integer or an mapping to integers coordinate values are
+    ignored. That is, the filter is applied even if the data points are not evenly
+    spaced. The resulting filtered data may thus have no meaningful interpretation.
+
+    Parameters
+    ----------
+    x:
+        Input variable or data array.
+    sigma:
+        Standard deviation for Gaussian kernel. The standard deviations of the Gaussian
+        filter are given as a mapping from dimension labels to numbers, or as a single
+        number, in which case it is equal for all axes.
+    order:
+        The order of the filter along each dimension, given as mapping from dimension
+        labels to integers, or as a single integer. An order of 0 corresponds to
+        convolution with a Gaussian kernel. A positive order corresponds to convolution
+        with that derivative of a Gaussian.
+
+    Examples
+    --------
+
+      >>> from scipp.ndimage import gaussian_filter
+      >>> da = sc.data.data_xy()
+
+    With sigma as integer:
+
+      >>> filtered = gaussian_filter(da, sigma=4)
+
+    With sigma based on input coordinate values:
+
+      >>> filtered = gaussian_filter(da, sigma=sc.scalar(1.2, unit='mm'))
+
+    With different sigma for different dimensions:
+
+      >>> filtered = gaussian_filter(da, sigma={'x':sc.scalar(1.2, unit='mm'), \
+                                                'y':sc.scalar(3.3, unit='mm')})
+    """
+    sigma = _positional_index(x, sigma, name='sigma', dtype=float)
+    order = order if isinstance(order, int) else [order[dim] for dim in x.dims]
```

Comment:
This should have the same dimension check as `_positional_index`.

Reply:
... so there are some missing tests ;)

Good catch!